keras_nlp/samplers/beam_sampler.py

```diff
@@ -162,16 +192,32 @@ def gather_beams(x):
         )
 
         # Gather the top beam at each batch index.
-        prompt, log_probs = unflatten_beams(prompt), unflatten_beams(log_probs)
-        top_beams = tf.math.argmax(log_probs, axis=-1)[:, tf.newaxis]
-        prompt = tf.gather(prompt, top_beams, axis=1, batch_dims=1)
-        return tf.squeeze(prompt, axis=1)
+        all_prompts, all_log_probs = unflatten_beams(prompt), unflatten_beams(
+            log_probs
+        )
+        top_beams = tf.math.argmax(all_log_probs, axis=-1)[:, tf.newaxis]
```

Comment:
These 2 lines are only useful in the branch `if not self.return_all_beams:`, we can fold them below to slightly improve our performance.